backend/dao/winesDAO.js

```diff
@@ -94,29 +140,71 @@ module.exports = class WinesDAO {
       const displayCursor = cursor
         .limit(winesPerPage)
         .skip(winesPerPage * page);
-      //console.log(displayCursor);
+
       try {
         const winesList = await displayCursor.toArray();
-        //console.log(winesList);
-        //!console.log("wine list: ", winesList);
+
         const totalWines = await wines.countDocuments(query);
         console.log("total wines:", totalWines);
+        //*tblshooting, empty the query here..
+        //todo: test more usecases for problems
+        query = {};
+
         return { winesList, totalWines };
       } catch (e) {
         console.error(`Unable to convert cursor to array: ${e}`);
         return { wineList: [], totalWines: 0 };
       }
     }
   }
-
-  static async addWine(req, res, next) {
+  //? ************************************* */
+  //* ************Add Wine***************** */
+  //* ************************************* */
+  static async addWine(req) {
     try {
       const wineDoc = req;
-      //console.log(wineDoc);
 
       return await wines.insertOne(wineDoc);
     } catch (e) {
       console.error(`Unabe to post wine: ${e}`);
     }
   }
+  //? ************************************* */
+  //* ************Single Wine************** */
+  //* ************************************* */
+  static async getSingleWine({
+    filter = null,
+    page = 0,
+    winesPerPage = 5,
+  } = {}) {
+    console.log("filter: ", filter);
+    if (filter) {
+      query.wine_id = { $eq: filter.id };
+    }
+
+    console.log("query:", query);
+    let cursor;
+    try {
+      cursor = await wines.find(query);
+    } catch (e) {
+      console.error(`Unable to issue find query: ${e}`);
+      return { singleWine: 0 };
+    }
+    const displayCursor = cursor.limit(winesPerPage).skip(winesPerPage * page);
+
+    try {
+      const winesList = await displayCursor.toArray();
+      console.log(winesList);
+      const totalWines = await wines.countDocuments(query);
+      console.log("total wines:", totalWines);
+      //*tblshooting, empty the query here..
+      //todo: test more usecases for problems
+      query = {};
+
+      return { winesList, totalWines };

```

Comment:
If the function is called `getSingleWine`, why do we return `winesList`?
Shouldn't we return a single wine object?

Reply:
It is single wine, but because the query is for wine_id, I set it up for error Incase there are multiple wine_ids- since I was not able to yet find out how to findById. Wine_id is human made, and I have it set up Incase there are repeat values. Ideally, it would only return one wine, but possible that more than one is sent in... is a replacement for until I figure out findById